src/App.tsx

```diff
@@ -1,10 +1,65 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Notification } from './components/notification/Notification';
+
+import { Todo } from './types/Todo';
+
+import { getTodos } from './api/todos';
+import { Footer } from './components/footer/Footer';
+import { Header } from './components/header/Header';
+import { TodoList } from './components/todoList/TodoList';
+
+const USER_ID = 6796;
 
-const USER_ID = 0;
+export enum SortType {
+  ALL = 'All',
+  ACTIVE = 'Active',
+  COMPLETED = 'Completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState('All');
+  const [isError, setIsError] = useState(true);
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (filterBy) {
+        case SortType.ALL:
+          return todos;
+
+        case SortType.ACTIVE:
+          return !todo.completed;
+
+        case SortType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return [];
+      }
+    })
+    );
+  }, [filterBy, todos]);
+
+  const noCompleteTodos = todos.some((todo) => !todo.completed);
+
+  const loadTodosData = async () => {
+    try {
+      setIsError(false);
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setIsError(true);
```

Comment:
specify error. use an enum of messages. it will be useful in the future, to give users understanding about what exactly went wrong